internal/backend/match.go

```diff
@@ -71,7 +71,7 @@ func match(ref, pattern, del, mailboxName string) (string, bool) {
 		return matchRoot(ref, del)
 	}
 
-	rx := fmt.Sprintf("^%v", regexp.QuoteMeta(ref+pattern))
+	rx := fmt.Sprintf("^%v", regexp.QuoteMeta(canon(ref+pattern, del)))
```

Comment:
Do you still need this here?

Reply:
Yes I do, the IMAP grammar unfortunately doesn't treat `pattern` here as a mailbox, meaning the parser won't handle the case for us. This also allows us to handle the edge case `tag list "INB" "ox"` properly.